orca-qos/src/main/kotlin/com/netflix/spinnaker/orca/qos/strategy/naive/NaivePromotionPolicy.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.qos.strategy.naive
+
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.qos.PromotionPolicy
+import com.netflix.spinnaker.orca.qos.PromotionResult
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnProperty("qos.strategy.naive.enabled")
+class NaivePromotionPolicy : PromotionPolicy {
+
+  override fun apply(candidates: List<Execution>) =
+   PromotionResult(
+     source = javaClass.simpleName,
+     candidates = candidates.subList(0, 1),
+     finalized = false,
+     reason = "Naive policy promotes 1 execution every cycle"
```

Comment:
For this particular case, `candidates` is currently (initial default) sorted list by `buildTime` right? 

So we're just naively promoting the most recently built execution (or maybe it's the oldest execution).

Also this is just the naive implementation, I wonder if it's worth being explicit about the sort order rather than just `subList(0, 1)` .. or maybe it just doesn't matter at all!

Reply:
It would promote the oldest execution in its current form (since all candidates are fist sorted, as you note). I figured pre-sorting the candidates by build time would likely save us some cycles if / when we get to a chain of promotion policies.

I think for a naive implementation (and likely one that won't live very long), this should suffice for now?

Reply:
I agree.

I'm mentally thinking through what it might look like to precalculate a score up-front so that you could store in a sorted set and avoid having to fetch the world.